src/core/GameRunner.ts

```diff
@@ -225,6 +225,23 @@ export class GameRunner {
       borderTiles: player.borderTiles(),
     } as PlayerBorderTiles;
   }
+  public attackAveragePosition(
+    playerID: number,
+    attackID: string,
+  ): Cell | null {
+    const player = this.game.playerBySmallID(playerID);
+    if (!player.isPlayer()) {
+      throw new Error(`player with id ${playerID} not found`);
+    }
+
+    const attacks = player.outgoingAttacks().concat(player.incomingAttacks());
+    const attack = attacks.find((a) => a.id() === attackID);
```

Comment:
concatenating these arrays is not necessary

```suggestion
    const condition = ((a) => a.id() === attackID);
    const attack =
      player.outgoingAttacks().find(condition) ??
      player.incomingAttacks().find(condition);
```